code/controllers/subsystem/input.dm

```diff
@@ -9,6 +9,24 @@ SUBSYSTEM_DEF(input)
 
 	var/list/macro_set
 
+	///what mobs are currently queued to click an atom. ran through every tick before movement input is processed.
+	///list verb callbacks storing information for clicks
+	var/list/datum/callback/verb_callback/queued_clicks = list()
+	///running average of how many clicks inputted by a player the server processes every second. used for the subsystem stat entry
+	var/clicks_per_second = 0
+	///count of how many clicks onto atoms have elapsed before being cleared by fire(). used to average with clicks_per_second.
+	var/current_clicks = 0
+	///acts like clicks_per_second but only counts the clicks actually processed by SSinput itself while clicks_per_second counts all clicks
+	var/delayed_clicks_per_second = 0
+	///running average of how many movement iterations from player input the server processes every second. used for the subsystem stat entry
+	var/movements_per_second = 0
+	///running average of the amount of real time clicks take to truly execute after the command is originally sent to the server.
+	///if a click isnt delayed at all then it counts as 0 deciseconds.
+	var/average_click_delay = 0
+
+	///if for whatever reason clicks arent being executed fast enough set this to TRUE and all clicks will immediately execute
+	var/FOR_ADMINS_IF_CLICKS_BROKE_immediately_execute_all_clicks = FALSE
```

Comment:
Why do this in the input subsystem exactly? cause it's on brand?

Reply:
Clicks are the only client input I can think of where latency matters more than resolving movement